### PR TITLE
`azurerm_private_endpoint` - support for `ip_configuration.member_name` property

### PR DESCRIPTION
**Pull request by wuxu92**
Cannot simply reuse `subresource_name` as `member_name` in `ip_configuration` property.(as described in the issue). The local testing response message says the default value is `default`, so I set the property's Default field as `default`. 

```
=== RUN   TestAccPrivateEndpoint_WithIPConfigForKeyVault
=== PAUSE TestAccPrivateEndpoint_WithIPConfigForKeyVault
=== CONT  TestAccPrivateEndpoint_WithIPConfigForKeyVault
--- PASS: TestAccPrivateEndpoint_WithIPConfigForKeyVault (437.53s)
PASS
```

image

Fixes: https://github.com/hashicorp/terraform-provider-azurerm/issues/18443

### CODE REVIEW

File: internal/services/network/private_endpoint_resource.go [hunk 1]
```diff
@@ -753,7 +760,7 @@ func expandPrivateEndpointIPConfigurations(input []interface{}) *[]network.Priva
 			PrivateEndpointIPConfigurationProperties: &network.PrivateEndpointIPConfigurationProperties{
 				PrivateIPAddress: utils.String(privateIPAddress),
 				GroupID:          utils.String(subResourceName),
-				MemberName:       utils.String(subResourceName),
+				MemberName:       utils.String(v["member_name"].(string)),
```
Comment (katbyte) [inline internal/services/network/private_endpoint_resource.go:763]: this is a breaking change for existing users/resourceS?

File: website/docs/r/private_endpoint.html.markdown [hunk 2]
```diff
@@ -199,7 +199,9 @@ An `ip_configuration` supports the following:
 
 * `private_ip_address` - (Required) Specifies the static IP address within the private endpoint's subnet to be used. Changing this forces a new resource to be created.
 
-* `subresource_name` - (Required) Specifies the subresource this IP address applies to. `subresource_names` corresponds to `group_id` and in this context is also used for `member_name`. Changing this forces a new resource to be created.
+* `subresource_name` - (Required) Specifies the subresource this IP address applies to. `subresource_names` corresponds to `group_id`. Changing this forces a new resource to be created.
+
+* `member_name` - (Optional) Specifies the member name of this IP address applies to. `member_name` corresponds to `group_id`, the default value is `default`.
```
Comment (katbyte) [inline website/docs/r/private_endpoint.html.markdown:204]: if `member_name` corresponds to `group_id` why is the property not called group_id?

File: website/docs/r/private_endpoint.html.markdown [hunk 3]
```diff
@@ -199,7 +199,9 @@ An `ip_configuration` supports the following:
 
 * `private_ip_address` - (Required) Specifies the static IP address within the private endpoint's subnet to be used. Changing this forces a new resource to be created.
 
-* `subresource_name` - (Required) Specifies the subresource this IP address applies to. `subresource_names` corresponds to `group_id` and in this context is also used for `member_name`. Changing this forces a new resource to be created.
+* `subresource_name` - (Required) Specifies the subresource this IP address applies to. `subresource_names` corresponds to `group_id`. Changing this forces a new resource to be created.
```
Comment (katbyte) [inline website/docs/r/private_endpoint.html.markdown:202]: and this also is group id?

**Review by katbyte — CHANGES_REQUESTED** [review]

File: internal/services/network/private_endpoint_resource.go (hunk 1 above)
Comment (wuxu92) [reply internal/services/network/private_endpoint_resource.go]: i updated the code to not break current logic

File: website/docs/r/private_endpoint.html.markdown (hunk 3 above)
Comment (wuxu92) [reply website/docs/r/private_endpoint.html.markdown:202]: by swagger, the `subresource_name`  should be alias to group_id [specs](https://github.com/Azure/azure-rest-api-specs/blob/b41f929626289b59e31be8a1091c99994864b096/specification/network/resource-manager/Microsoft.Network/stable/2021-05-01/privateEndpoint.json#L1019), and member_name is the name of the member of group_id which should not reuse the `subresource_name` value.

Comment (katbyte) [reply website/docs/r/private_endpoint.html.markdown:202]: i'm sorry but i don't follow, could you try and explain that again? i don't see subresource_name there at all even

Comment (wuxu92) [reply website/docs/r/private_endpoint.html.markdown:202]: there are three properties `privateIPAddress`, `memberName` and `groupId` defined in swagger for `PrivateEndpointIPConfigurationProperties`, but only two properties defined in `azurerm` in the first place, they are `private_ip_address` (for `privateIPAddress`) and `subresource_name` (for both `groupId` and `memberName`). I believe the `subresource_name` defined in the azurerm schema is a rename of `groupId`. this PR is adding a new property for `memberName` to distinguish `groupId` and `memberName` in the request.

**Comment by gl322** [comment]
Hi folks,

first of all thank you for taking care of that issue. I had also a issue with deploying a private endpoint for a ACR and had a similar approach as @wuxu92 and it was working for deploying a private endpoint for a service which has only one ipaddress. But ACR has two, after changing max items for the ipconfiguration it was also working fine. As private enpoint can also used for a private link service and it has a up to 8 ip address it would may be make sense to set MaxItems to 8. What do you think? 
Thanks and best regards

**Comment by wuxu92** [comment]
@gl322 I have changed the MaxItem of `private_service_connection` up to 8 in this PR. @katbyte is there anything I should do to make this PR cloud merged?

**Comment by tombuildsstuff** [comment]
@wuxu92 can you explain the use-case for this one?

**Comment by wuxu92** [comment]
> @wuxu92 Xu Wu FTE can you explain the use-case for this one?

this pr fixes https://github.com/hashicorp/terraform-provider-azurerm/issues/18443 in some cases, the `subresource_name` and the `member_name` use the different values

**Comment by Juwebien** [comment]
Thanks @wuxu92, really need this feature.

I hope a quick merge :)

**Comment by wuxu92** [comment]
@tombuildsstuff how do you think of the PR now?, is there anything I need to do to make it merged?

**Comment by wuxu92** [comment]
there is a new PR to fix this issue too: https://github.com/hashicorp/terraform-provider-azurerm/pull/19389

**Comment by katbyte** [comment]
closing in favour of #19389

**Pull request closed without merging**